### PR TITLE
fix: address mismatch in ids passed to cbr rules

### PR DESCRIPTION
**Pull request by vburckhardt**
### Description

fixes https://github.com/terraform-ibm-modules/terraform-ibm-cos/issues/201

#### No release required

- [ ] Examples or tests (addition or updates of examples or tests)
- [ ] Documentation update
- [ ] CI-related update (pipeline, etc.)
- [ ] Other changes that don't affect Terraform code

#### Release required

- [x] Bug fix (patch release (`x.x.X`): Change that fixes an issue and is compatible with earlier versions)
- [ ] New feature (minor release (`x.X.x`): Change that adds functionality and is compatible with earlier versions)
- [ ] Breaking change (major release (`X.x.x`): Change that is likely incompatible with previous versions)

##### Release notes content

Replace this text with information that users need to know about the bug fixes, features, and breaking changes. This information helps the merger write the commit message that is published in the release notes for the module.

---

### Checklist for reviewers

- [x] The PR references a GitHub issue.
- [ ] If relevant, a test for the change is included or updated with this PR.
- [ ] If relevant, documentation for the change is included or updated with this PR.

### Merge actions for mergers

- Merge by using "Squash and merge".
- Use a relevant [conventional commit](https://www.conventionalcommits.org/) message that is based on the PR contents and any release notes provided by the PR author.

    The commit message determines whether a new version of the module is needed, and if so, which semver increment to use (major, minor, or patch).

### CODE REVIEW

**Comment by vburckhardt** [comment]
Generated rules with the fix (using the 'complete' example - https://github.com/terraform-ibm-modules/terraform-ibm-cos/tree/main/examples/complete)
image

File: tests/pr_test.go [hunk 1]
```diff
@@ -55,6 +55,12 @@ func setupOptions(t *testing.T, prefix string, dir string) *testhelper.TestOptio
 		Prefix:        prefix,
 		ResourceGroup: resourceGroup,
 		Region:        region,
+		// Ignore updates to CBR rule - see https://github.com/terraform-ibm-modules/terraform-ibm-cos/issues/201
+		IgnoreUpdates: testhelper.Exemptions{List: []string{
+			"module.cos_bucket1.module.instance_cbr_rule[0].ibm_cbr_rule.cbr_rule",
+			"module.cos_bucket1.module.bucket_cbr_rule[0].ibm_cbr_rule.cbr_rule",
+			"module.cos_bucket2.module.bucket_cbr_rule[0].ibm_cbr_rule.cbr_rule",
+		}},
```
Comment (ocofaigh) [inline tests/pr_test.go:63]: @vburckhardt Is this a one off ignore to get the upgrade test to pass? If so, lets just skip the upgrade for this PR, and remove this ignore code so it does remain for future PRs?

Comment (vburckhardt) [reply tests/pr_test.go:63]: Yes it's a once off. I think this is fine to leave as is for now.

**Comment by vburckhardt** [comment]
@jojustin @daniel-butler-irl @ocofaigh - Should be good to go whenever you have final cycles for review.

**Review by ocofaigh — APPROVED** [review]

**Pull request merged**